presto-hive/src/main/java/com/facebook/presto/hive/s3/PrestoS3FileSystem.java

```diff
@@ -546,8 +545,13 @@ private boolean deleteObject(String key)
     @Override
     public boolean mkdirs(Path f, FsPermission permission)
     {
-        // no need to do anything for S3
-        return true;
+        try {
+            s3.putObject(getBucketName(uri), keyFromPath(f) + PATH_SEPARATOR, "");
```

Comment:
I have a concern that there may be code which relies on an assumption that `mkdirs` in S3 is a low-overhead operation within Presto. Have you done any other testing in a real deployment connected to S3 to see if there are performance regressions due to this change?